web/ui/react-app/src/components/SearchBar.tsx

```diff
@@ -1,29 +1,35 @@
-import React, { ChangeEvent, FC } from 'react';
+import React, { ChangeEvent, FC, useEffect } from 'react';
 import { Input, InputGroup, InputGroupAddon, InputGroupText } from 'reactstrap';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faSearch } from '@fortawesome/free-solid-svg-icons';
 
 export interface SearchBarProps {
-  handleChange: (e: ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => void;
+  handleChange: (e: string) => void;
   placeholder: string;
+  defaultValue: string;
 }
 
-const SearchBar: FC<SearchBarProps> = ({ handleChange, placeholder }) => {
+const SearchBar: FC<SearchBarProps> = ({ handleChange, placeholder, defaultValue }) => {
   let filterTimeout: NodeJS.Timeout;
 
   const handleSearchChange = (e: ChangeEvent<HTMLTextAreaElement | HTMLInputElement>) => {
     clearTimeout(filterTimeout);
     filterTimeout = setTimeout(() => {
-      handleChange(e);
+      handleChange(e.target.value);
     }, 300);
   };
 
+  useEffect(() => {
+    handleChange(defaultValue);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Hmmmmm. Ideally we wouldn't disable this ESLint warning here, and rather pass in the necessary `handleChange` as a dependency as well. However, that callback function is created anew in the parent component on every render, so it would cause a deps change here on ever render of the parent. To prevent that, we would have to create the callback function in the parent using `useCallback` (https://reactjs.org/docs/hooks-reference.html#usecallback). I think we should either do that or go back to just having `[]` as a deps array (which I think will always work without warnings).

Reply:
The difference between useMemo and useCallback is that one returns a function and the other a value, when I place the dependency on useEffect of handleChange, the implementation of the expr in url from the SearchBar on the page is damaged. This is because a loop is created apparently, what possible solution apart from disabling the eslint warning do you see?

Reply:
do you mean this type of change?
![image](https://user-images.githubusercontent.com/48663717/165372458-9415af61-093d-4392-ae47-babe78f64275.png)

Reply:
sorry I had not understood you, thank you very much I upload the changes :)